interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -359,28 +359,31 @@ public void pingPong() throws Exception {
             .build());
 
     @SuppressWarnings("unchecked")
-    StreamObserver<StreamingOutputCallResponse> responseObserver = mock(StreamObserver.class);
+    StreamRecorder<StreamingOutputCallResponse> responseObserver = StreamRecorder.create();
     StreamObserver<StreamingOutputCallRequest> requestObserver
         = asyncStub.fullDuplexCall(responseObserver);
     for (int i = 0; i < requests.size(); i++) {
       requestObserver.onNext(requests.get(i));
-      verify(responseObserver, timeout(operationTimeoutMillis())).onNext(goldenResponses.get(i));
```

Comment:
I added comments about the implementation. However, it may be easier to use `BlockingQueue<Object>` (like `LinkedBlockingQueue`). You can create your own `StreamObserver` that adds any `StreamingOutputCallResponse` or `Throwable` to the queue. That would also naturally produce a reasonable error message if an entry isn't what you expect. See `serverStreamingShouldBeFlowControlled()` for an example.